### PR TITLE
feat: annotations are now only shown in the cell they're created in

### LINKED ISSUE #1037: Annotation In Cell: Only Render annotations that match the current cell id
### Background
We currently render visible annotations in the `default` stream. This will need to change based on the need to render annotations in the cell they were written.

### Acceptance Criteria
- Filter annotations such that only annotation in the stream that matches the current cell id are rendered

### LINKED ISSUE #1038: Annotation In Cell: Write the current cell id as the stream name
### Background
Background - we currently write all annotations into the `default` stream. We'll instead want to write it to the stream that corresponds to the cell id the annotation is being rendered in.

### Acceptance Criteria
- The cell id is passed to the API as the stream name

### LINKED ISSUE #1039: Annotation In Cell: Make sure editing and deleting works with the new data shape
### Background
Editing currently depends on the soon-to-be-outdated assumption that we use the default stream

### Acceptance Criteria
- Editing works smoothly in the new stream model

### LINKED ISSUE #1069: Annotations: Remove annotation tab from settings page
### Background
The annotations Index page isn't rendered, but a link to it is still accessible from the Settings page. Remove this link.

### PR DESCRIPTION
**Pull request by hoorayimhelping**
Closes #1037 
Closes #1038 
Closes #1039 
Closes #1069

This was a coast-to-coast effort. Jill and Zoe worked on it then handed it off to Zoe, then Zoe worked on it and handed it off to me.

Screen Shot 2021-03-31 at 4 46 41 PM

### CODE REVIEW

File: src/annotations/reducers/index.ts [hunk 1]
```diff
@@ -57,41 +57,33 @@ export const annotationsReducer = (
         }),
       }
     }
-    case ENABLE_ANNOTATION_STREAM: {
```
Comment (jrenee42) [inline src/annotations/reducers/index.ts:60]: need to delete the imports too!

File: src/visualization/types/Graph/view.tsx [hunk 2]
```diff
@@ -80,14 +86,10 @@ const XYPlot: FC<Props> = ({properties, result, timeRange, annotations}) => {
...
-  // it doesn't know that it is even *in* a dashboard, much less which dashboard it is in.
-  // so having annotations on per dashboard is not supported yet
-  // these next two values are set in the dashboard, and used whether or not this view is in a dashboard
-  // or in configuration/single cell popout mode
-  // would need to add the annotation control bar to the VEOHeader to get access to the controls,
-  // which are currently global values, not per dashboard
   const inAnnotationWriteMode = useSelector(isSingleClickAnnotationsEnabled)
-  const visibleAnnotationStreams = useSelector(getVisibleAnnotationStreams)
+
+  //use cellid as the annotation 'stream'
```
Comment (jrenee42) [inline src/visualization/types/Graph/view.tsx:91]: need a space after the // for eslint

**Review by jrenee42 — CHANGES_REQUESTED** [review]

File: src/types/annotation.ts [hunk 3]
```diff
@@ -1,39 +1,47 @@
...
 }
 
-export interface AnnotationResponse {
+export interface EditAnnotation {
+  id: string
+  message: string
+  startTime: string
   stream: string
+  summary: string
+}
```
Comment (hoorayimhelping) [inline src/types/annotation.ts:17]: this was moved from `EditAnnotationForm`, everything else just got alphabetized

**Comment by hercules-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 35 | :red_circle:
accessibility | 96 | :green_circle:
best-practices | 93 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :question:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?vars%5Blighthouse_score_branch%5D=1064)

File: src/visualization/types/Graph/view.tsx [hunk 4]
```diff
@@ -80,18 +84,9 @@ const XYPlot: FC<Props> = ({properties, result, timeRange, annotations}) => {
   )
   const dispatch = useDispatch()
 
-  // it doesn't know that it is even *in* a dashboard, much less which dashboard it is in.
-  // so having annotations on per dashboard is not supported yet
-  // these next two values are set in the dashboard, and used whether or not this view is in a dashboard
-  // or in configuration/single cell popout mode
```
Comment (jrenee42) [inline src/visualization/types/Graph/view.tsx:86]: the first three lines of comments are definitely obsolete/not relevant any more,

but the last three lines references 'inAnnotationWriteMode", which is still relevant

**Review by jrenee42 — CHANGES_REQUESTED** [review]

File: src/annotations/actions/creators.ts [hunk 5]
```diff
@@ -2,7 +2,6 @@ import {Annotation, AnnotationResponse, AnnotationStream} from 'src/types'
 
 export const DELETE_ANNOTATION = 'DELETE_ANNOTATION'
 export const DISABLE_ANNOTATION_STREAM = 'DISABLE_ANNOTATION_STREAM'
```
[the remark this answers is not in the document]
Comment (jrenee42) [reply src/annotations/actions/creators.ts]: (i just pushed up a change to remove it)

File: src/visualization/types/Graph/view.tsx (hunk 4 above)
Comment (jrenee42) [reply src/visualization/types/Graph/view.tsx:86]: restored the relevant comments; and cleaned up the commits a little (made them all semantic; squashed two of them)

Comment (hoorayimhelping) [reply src/visualization/types/Graph/view.tsx:86]:
```
// it doesn't know that it is even *in* a dashboard, much less which dashboard it is in.
// so having annotations on per dashboard is not supported yet
```

isn't really true. every cell operation happens in the context of a dashboard, so it's not that it does or doesn't know which dashboard it's in, it's just that it's irrelevant. it's not that per dashboard annotations aren't supported, it's that we chose not to do them in favor of cell-only annotations

Comment (jrenee42) [reply src/visualization/types/Graph/view.tsx:86]: those two lines were not restored (because I agree that part is not relevant)

**Review by jrenee42 — APPROVED** [review]
approved; bucky please merge after your local tests pass!

**Pull request merged**